python/cudf/cudf/core/indexing.py

```diff
@@ -479,3 +495,14 @@ def _normalize_dtypes(df):
         for name, col in df._data.items():
             df[name] = col.astype(normalized_dtype)
     return df
+
+
+def _append_new_row_inplace(col: ColumnLike, value: ScalarLike):
```

Comment:
I wonder if this would be useful as a method of our `ColumnBase` class.

Reply:
I think `ColumnBase` has a `append` function, which I hope to avoid here to reduce overhead.